File: native/mixdog-graph/tests/fixtures/calls/csharp/sample.cs
// hidden()
string quoted = "hidden()";

plain();

void run(Chain a) {
  inner();
  nest(leaf(1));
  a.b().c();
}

string mark = "μ"; seed();
new Widget();

void inner() {}
int leaf(int x) => x;
void nest(int x) {}
void helper() {}
void seed() {}
int plain() => 0;

class Chain {
  public Chain b() => this;
  public Chain c() => this;
}

class Widget {
  public void ping() {}
  public void act() {
    helper();
    this.ping();
  }
}

File: native/mixdog-graph/tests/fixtures/calls/zig/sample.zig
// hidden()
const quoted = "hidden()";

fn inner() void {}
fn leaf(x: i32) i32 {
    return x;
}
fn nest(x: i32) void {
    _ = x;
}
fn helper() void {}
fn seed() void {}
fn plain() void {}

const Chain = struct {
    fn b(self: Chain) Chain {
        return self;
    }
    fn c(self: Chain) Chain {
        return self;
    }
};

fn run(a: Chain) void {
    inner();
    nest(leaf(1));
    _ = a.b().c();
}

const Widget = struct {
    fn ping(self: Widget) void {
        _ = self;
    }
    fn act(self: Widget) void {
        helper();
        self.ping();
    }
    fn init() Widget {
        return .{};
    }
};

comptime {
    plain();
}

pub fn main() void {
    const mark = "μ"; seed();
    _ = mark;
    _ = quoted;
    _ = Widget.init();
}
